### PR TITLE
Optimize meta-label definition and usage guidelines

### PR DESCRIPTION
**Pull request by remyroche**
Implements a comprehensive meta-labeling framework with dynamic scoring, relevance evaluation, and mixture-of-experts weighting to optimize specialist model integration.

This PR addresses the user's request for guidelines to optimize meta-label definition and use. It provides concrete mechanisms for defining when a meta-label applies (gating), scoring its intensity, confidence, and reliability, assessing its relevance (using MI, SHAP, Sharpe), and dynamically weighting specialized models against a general model via a calibrated Mixture-of-Experts (MoE) approach. This allows for more intelligent and adaptive use of meta-labels in the system.

---
<a href="https://cursor.com/background-agent?bcId=bc-c158693d-640b-42b0-99b9-d038c46edfb7">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/open-in-cursor-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://cursor.com/open-in-cursor-light.svg">
    Open in Cursor
  </picture>
</a>
<a href="https://cursor.com/agents?id=bc-c158693d-640b-42b0-99b9-d038c46edfb7">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/open-in-web-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://cursor.com/open-in-web-light.svg">
    Open in Web
  </picture>
</a>

### CODE REVIEW

File: src/analyst/meta_label_relevance.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            from sklearn.feature_selection import mutual_info_classif
+
+            mi = mutual_info_classif(Xn.fillna(0.0), y.astype(int), random_state=random_state)
+        else:
+            from sklearn.feature_selection import mutual_info_regression
+
+            mi = mutual_info_regression(Xn.fillna(0.0), y, random_state=random_state)
+        return {c: float(v) for c, v in zip(Xn.columns, mi)}
+    except Exception:
+        return {}
```
[the remark this answers is not in the document]
Comment (remyroche) [reply src/analyst/meta_label_relevance.py]: @cursoragent fix

File: src/analyst/ml_confidence_predictor.py [hunk 2]
```diff
@@ -3066,6 +3066,47 @@ async def predict_label_confidences(
...
+                hi = w_max if w_max < 1.0 else 1.0
+                w = float(np.clip(s, lo, hi))
+            else:
+                w = 0.0
+            weights[label] = w
+        if normalize:
+            total = float(sum(weights.values()))
+            if total > 0:
+                weights = {k: float(v / total) for k, v in weights.items()}
+        return weights
```
[the remark this answers is not in the document]
Comment (remyroche) [reply src/analyst/ml_confidence_predictor.py:3108]: @cursoragent fix
